<?xml version="1.0"?>
<launch>
    <group ns="stereo">
		<arg name="node_name" default="norlab_stereo_driver_node" />
		<arg name="startup_user_set" default="UserSet1" />
		<arg name="image_encoding" default="bayer_rggb8" />
		<arg name="enable_bracketing" default="true" />
		<arg name="frame_rate" default="10" />
		<arg name="camera1_calibration_url" default="package://norlab_basler_camera_driver/config/camera1_calibration_file.yaml" />
		<arg name="camera2_calibration_url" default="package://norlab_basler_camera_driver/config/camera2_calibration_file.yaml" />

		<arg name="respawn" default="false" />
		<arg name="debug" default="false" />
		<arg unless="$(arg debug)" name="launch_prefix" value="" />
    	<arg if="$(arg debug)" name="launch_prefix" value="gdb -ex run --args" />

    	<node name="$(arg node_name)" pkg="norlab_stereo_driver_node" type="norlab_stereo_driver_node" output="screen"
          	respawn="$(arg respawn)" launch-prefix="$(arg launch_prefix)">
        	<param name="startup_user_set" value="$(arg startup_user_set)"/>
			<param name="image_encoding" value="$(arg image_encoding)"/>
			<param name="enable_bracketing" value="$(arg enable_bracketing)"/>
			<param name="frame_rate" value="$(arg frame_rate)"/>
			<param name="camera1_calibration_url" value="$(arg camera1_calibration_url)"/>
			<param name="camera2_calibration_url" value="$(arg camera2_calibration_url)"/>
			<rosparam param="bracketing_values">[1000.0, 2000.0, 4000.0, 8000.0, 16000.0, 32000.0]</rosparam>
    	</node>
    </group>

	<!-- DEBAYER and RECTIFY-->
	<arg name="enable_debayer" default="false" />
	<arg name="enable_rectify" default="false" />

	<!-- CAMERA 1 -->
	<node if="$(arg enable_debayer)" name="stereo_camera1_manager" pkg="nodelet" type="nodelet" args="manager" output="screen" ns="stereo">
		<param name="num_worker_threads" value="3" type="int"/>
	</node>
	<node if="$(arg enable_debayer)" pkg="nodelet" type="nodelet" name="stereo_debayer_camera1" ns="stereo" args="load image_proc/debayer stereo_camera1_manager">
		<param name="bayer_pattern" value="rggb"/>
		<remap from="image_raw" to="/stereo/camera1/image"/>
		<remap from="camera_info" to="/stereo/camera1/camera_info"/>
		<remap from="image_color" to="/stereo/camera1/image_debayer_color"/>
		<remap from="image_mono" to="/stereo/camera1/image_debayer_mono"/>
	</node>
	<node if="$(arg enable_rectify)" pkg="nodelet" type="nodelet" name="stereo_rectify_camera1" ns="stereo" args="load image_proc/rectify stereo_camera1_manager">
		<param name="bayer_pattern" value="rggb"/>
		<remap from="image_mono" to="/stereo/camera1/image_debayer_color"/>
		<remap from="camera_info" to="/stereo/camera1/camera_info"/>
		<remap from="image_rect" to="/stereo/camera1/image_recitify_color"/>
	</node>

	<!-- CAMERA 2 -->
	<node if="$(arg enable_debayer)" name="stereo_camera2_manager" pkg="nodelet" type="nodelet" args="manager" output="screen" ns="stereo">
		<param name="num_worker_threads" value="3" type="int"/>
	</node>
	<node if="$(arg enable_debayer)" pkg="nodelet" type="nodelet" name="stereo_debayer_camera2" ns="stereo" args="load image_proc/debayer stereo_camera2_manager">
		<param name="bayer_pattern" value="rggb"/>
		<remap from="image_raw" to="/stereo/camera2/image"/>
		<remap from="camera_info" to="/stereo/camera2/camera_info"/>
		<remap from="image_color" to="/stereo/camera2/image_debayer_color"/>
		<remap from="image_mono" to="/stereo/camera2/image_debayer_mono"/>
	</node>
	<node if="$(arg enable_rectify)" pkg="nodelet" type="nodelet" name="stereo_rectify_camera2" ns="stereo" args="load image_proc/rectify stereo_camera2_manager">
		<param name="bayer_pattern" value="rggb"/>
		<remap from="image_mono" to="/stereo/camera2/image_debayer_color"/>
		<remap from="camera_info" to="/stereo/camera2/camera_info"/>
		<remap from="image_rect" to="/stereo/camera2/image_recitify_color"/>
	</node>

</launch>
